app/admin/views.py

```diff
@@ -49,10 +49,16 @@ def invite_user():
     """Invites a new user to create an account and set their own password."""
     form = InviteUserForm()
     if form.validate_on_submit():
-        user = User(role=form.role.data,
-                    first_name=form.first_name.data,
-                    last_name=form.last_name.data,
-                    email=form.email.data)
+        role_choice = form.role.data.name
+        print role_choice
+        if role_choice == 'Vendor':
+            user = Vendor(email=form.email.data)
+        elif role_choice == 'Merchant':
+            user = Merchant(email=form.email.data)
+        else:
+            user = User(role=form.role.data,
```

Comment:
Can you add these options to the form dropdown? And I think while we're working on this code we should remove the `else` option, add a third option for the `Admin` role, and refuse to make a user if any other role form data is given to us